core/rubygems/version.rbs

```diff
@@ -167,7 +167,7 @@ module Gem
     # -->
     # True if the `version` string matches RubyGems' requirements.
     #
-    def self.correct?: (_ToS version) -> bool
+    def self.correct?: (_ToS & Kernel version) -> bool
```

Comment:
https://github.com/ruby/ruby/blob/9ee58b2054c1bbe722ae5a2a4ec6a750ee583220/lib/rubygems/version.rb#L174

`nil?` is used.